Events/2024-12-virtual-workshop-on-multiproject-cicd.md

```diff
@@ -0,0 +1,93 @@
+# Virtual Workshop on Multiproject CI/CD
+
+- Abstract submission deadline: November 22, 2024
+- Registration deadline: December 5, 2024
+- Event Date: December 6, 2024
+- Location: online
+- Website: https://multiprojectdevops.github.io/workshops/virtual_workshop2/
+- Organizers: Ryan M. Richard
+
+#### Contributed by: [Ryan M. Richard](https://github.com/ryanmrichard)
+
+#### Publication date: August 16, 2024
+
+<!-- deck text start -->
+This workshop continues efforts to build a community of practice for developing and maintaining CI/CD workflows for multiple projects.
+<!-- deck text ends -->
+
+Event Information | Details
+:--- | :---
+Event Name | [Virtual Workshop on Multiproject CI/CD](https://multiprojectdevops.github.io/workshops/virtual_workshop1/)
+Abstract submission deadline | November 22, 2024
+Registration deadline | December 5, 2024
+Event Date | December 6, 2024
+Website | https://multiprojectdevops.github.io/workshops/virtual_workshop2/
+
+
+### Summary
+
+The Multi-ProjectDevOps project was founded to establish best practices for
+decreasing the time needed to deploy multiple high-quality scientific software
+packages, particularly when the packages share common requirements. Key to
```

Comment:
@ryanmrichard I may be completely misunderstanding but when I saw *"multi-project", my mind went to *interdependency* of the packages...I mean like he kind of CI/CD which tests the packages working together instead of each in isolation? But, it sounds like this is really about defining a CI/CD workflow where maybe the same `workflow.yaml` file can be used and shared among a number of different packages with similar testing processes.

Reply:
@markcmiller86 In my experience it starts as just trying to share a file, but then it quickly gets harder. Package interdependency definitely plays a role. For example, I maintain a plugin framework and a series of plugins for that framework. Roughly speaking they all build the software the same way, test the software the same way, deploy the documentation the same way, etc. Each plugin depends on the framework and may also depend on other plugins in the ecosystem. While the plugins can be unit tested in isolation, integration/acceptance testing often requires other plugins. We handle the literal dependency management at the build system level (I'm hoping to move to a package manager rather than native CMake, but haven't had funding to hook that up), but you can also imagine having CI/CD tasks that make sure that the newest changes to a plugin doesn't break other plugins, and/or triggering a new deployment with all bleeding-edge dependencies.